python/eos/plot/plotter.py

```diff
@@ -374,80 +361,75 @@ class Observable(BasePlot):
            }
         """)
 
+        from .specs import Specs, ParametersSpec, KinematicsSpec,\
+        RangeSpec, ObservableSpec, VariableSpec, ParameterFileSpec
+
+        specs_prototype = Specs([
+            ObservableSpec(name='observable'),
+            VariableSpec(name='variable'),
+            KinematicsSpec(name='kinematics', optional=True, default={}),
+            ParametersSpec(name='parameters', optional=True, default={}),
+            # ParameterFileSpec(name='parameter-file', optional=True, default=None)
+        ])
+
         def __init__(self, plotter, item):
             super().__init__(plotter, item)
+            self.specs = deepcopy(self.specs_prototype)
 
         def plot(self):
-            item = self.item
-            oname = item['observable']
-
-            obs_entry = eos.Observables._get_obs_entry(oname)
-            valid_kin_vars = [kv for kv in obs_entry.kinematic_variables()]
-            eos.info('   plotting EOS observable "{}"'.format(oname))
+            item = self.item.copy()
 
-            # create kinematics
             kinematics = eos.Kinematics()
-            if 'kinematics' in item:
-                for k, v in item['kinematics'].items():
-                    if k not in valid_kin_vars:
-                        raise ValueError("Kinematic quantity '" + k + "' does not " +
-                        "match known ones for observable '" + oname + "': " + valid_kin_vars.__repr__())
-                    kinematics.declare(k, v)
-
-            # create parameters
             parameters = eos.Parameters.Defaults()
-            if 'parameters' in item and 'parameters-from-file' in item:
-                eos.warn('    overriding values read from \'parameters-from-file\' with explicit values in \'parameters\'')
-
-            if 'parameters-from-file' in item and type(item['parameters-from-file']) is str:
-                eos.warn('    overriding parameters from file')
-                parameters.override_from_file(item['parameters-from-file'])
-
-            if 'parameters' in item and type(item['parameters']) is dict:
-                for key, value in item['parameters'].items():
-                    parameters.set(key, value)
-
-            # create (empty) options
             options = eos.Options()
 
+            # process user input
+            # need oname for some specs
+            try:
+                oname = item['observable']
+            except KeyError:
+                raise ValueError("Missing key: 'observable'")
+            self.specs['variable'].observable_name = oname
+            self.specs['kinematics'].observable_name = oname
+            self.specs['parameters'].observable_name = oname
+
+            self.specs.validate_set(item)
```

Comment:
Why don't we validate in the ``__init__``? Consider the case that you modify your plot file and make a mistake.
Now you run through half an hour of plotting, before you hit your mistake at the end of the plot file. Not optimal...

This is why we distinguish between the ``__init__`` part and the ``plot`` method.